{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SwaggerClient(https://www.cbioportal.org/api)\n"
     ]
    }
   ],
   "source": [
    "from bravado.client import SwaggerClient\n",
    "\n",
    "cbioportal = SwaggerClient.from_url('https://www.cbioportal.org/api/v2/api-docs',\n",
    "                                config={\"validate_requests\":False,\"validate_responses\":False,\"validate_swagger_spec\":False})\n",
    "print(cbioportal)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "for a in dir(cbioportal):\n",
    "    cbioportal.__setattr__(a.replace(' ', '_').lower(), cbioportal.__getattr__(a))# "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancerTypeId = \"cllsll\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cll = cbioportal.Cancer_Types.getCancerTypeUsingGET(cancerTypeId).result()\n",
    "# print(cll)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['fetchStudiesUsingPOST',\n",
       " 'getAllStudiesUsingGET',\n",
       " 'getStudyUsingGET',\n",
       " 'getTagsForMultipleStudiesUsingPOST',\n",
       " 'getTagsUsingGET']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dir(cbioportal.Studies)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_studies = cbioportal.Studies.getAllStudiesUsingGET().result()\n",
    "cll_studies = [s for s in all_studies if s.cancerTypeId == cancerTypeId] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[CancerStudy(allSampleCount=105, cancerType=None, cancerTypeId='cllsll', citation='Quesada et al. Nature Genetics 2011', cnaSampleCount=None, completeSampleCount=None, description='Whole-exome sequencing of 105 Chronic Lymphocytic Leukemia samples.', groups='PUBLIC', importDate='2024-12-04 11:30:41', massSpectrometrySampleCount=None, methylationHm27SampleCount=None, miRnaSampleCount=None, mrnaMicroarraySampleCount=None, mrnaRnaSeqSampleCount=None, mrnaRnaSeqV2SampleCount=None, name='Chronic lymphocytic leukemia (ICGC, Nature Genetics 2011)', pmid='22158541', publicStudy=True, readPermission=True, referenceGenome='hg19', rppaSampleCount=None, sequencedSampleCount=None, status=0, studyId='cllsll_icgc_2011', treatmentCount=None),\n",
       " CancerStudy(allSampleCount=160, cancerType=None, cancerTypeId='cllsll', citation='Landau et al. Cell 2013', cnaSampleCount=None, completeSampleCount=None, description='Whole-exome sequencing of 160 CLL tumor/normal pairs.', groups='', importDate='2024-12-05 13:26:10', massSpectrometrySampleCount=None, methylationHm27SampleCount=None, miRnaSampleCount=None, mrnaMicroarraySampleCount=None, mrnaRnaSeqSampleCount=None, mrnaRnaSeqV2SampleCount=None, name='Chronic Lymphocytic Leukemia (Broad, Cell 2013)', pmid='23415222', publicStudy=True, readPermission=True, referenceGenome='hg19', rppaSampleCount=None, sequencedSampleCount=None, status=0, studyId='lcll_broad_2013', treatmentCount=None),\n",
       " CancerStudy(allSampleCount=506, cancerType=None, cancerTypeId='cllsll', citation='Puente et al. Nature 2015', cnaSampleCount=None, completeSampleCount=None, description='Mutation data from whole-genome or whole-exome sequencing of 428 CLL, 54 MBL, and 24 SLL tumor/normal pairs.', groups='', importDate='2024-12-20 02:53:22', massSpectrometrySampleCount=None, methylationHm27SampleCount=None, miRnaSampleCount=None, mrnaMicroarraySampleCount=None, mrnaRnaSeqSampleCount=None, mrnaRnaSeqV2SampleCount=None, name='Chronic Lymphocytic Leukemia (IUOPA, Nature 2015)', pmid='26200345', publicStudy=True, readPermission=True, referenceGenome='hg19', rppaSampleCount=None, sequencedSampleCount=None, status=0, studyId='cll_iuopa_2015', treatmentCount=None),\n",
       " CancerStudy(allSampleCount=537, cancerType=None, cancerTypeId='cllsll', citation='Landau et al. Nature 2015', cnaSampleCount=None, completeSampleCount=None, description='Whole exome sequencing of 537 chronic lymphocytic leukemia tumor/normal pairs.', groups='PUBLIC', importDate='2024-12-20 04:09:27', massSpectrometrySampleCount=None, methylationHm27SampleCount=None, miRnaSampleCount=None, mrnaMicroarraySampleCount=None, mrnaRnaSeqSampleCount=None, mrnaRnaSeqV2SampleCount=None, name='Chronic Lymphocytic Leukemia (Broad, Nature 2015)', pmid='26466571', publicStudy=True, readPermission=True, referenceGenome='hg19', rppaSampleCount=None, sequencedSampleCount=None, status=0, studyId='cll_broad_2015', treatmentCount=None),\n",
       " CancerStudy(allSampleCount=1154, cancerType=None, cancerTypeId='cllsll', citation='Knisbacher et al. Nature 2022', cnaSampleCount=None, completeSampleCount=None, description='Whole genome and whole exome sequencing of 1,154 samples from 1,148 patients with Chronic Lymphocytic Leukemia and monoclonal B cell lymphocytosis and their matched normals.', groups='', importDate='2024-12-20 11:07:33', massSpectrometrySampleCount=None, methylationHm27SampleCount=None, miRnaSampleCount=None, mrnaMicroarraySampleCount=None, mrnaRnaSeqSampleCount=None, mrnaRnaSeqV2SampleCount=None, name='Chronic Lymphocytic Leukemia (Broad, Nature Genetics 2022)', pmid='35927489', publicStudy=True, readPermission=True, referenceGenome='hg19', rppaSampleCount=None, sequencedSampleCount=None, status=0, studyId='cll_broad_2022', treatmentCount=None)]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cll_studies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cllsll_icgc_2011 Whole-exome sequencing of 105 Chronic Lymphocytic Leukemia samples. 105\n",
      "lcll_broad_2013 Whole-exome sequencing of 160 CLL tumor/normal pairs. 160\n",
      "cll_iuopa_2015 Mutation data from whole-genome or whole-exome sequencing of 428 CLL, 54 MBL, and 24 SLL tumor/normal pairs. 506\n",
      "cll_broad_2015 Whole exome sequencing of 537 chronic lymphocytic leukemia tumor/normal pairs. 537\n",
      "cll_broad_2022 Whole genome and whole exome sequencing of 1,154 samples from 1,148 patients with Chronic Lymphocytic Leukemia and monoclonal B cell lymphocytosis and their matched normals. 1154\n",
      "Total samples: 2462\n"
     ]
    }
   ],
   "source": [
    "total_samples = 0\n",
    "for study in cll_studies:\n",
    "    total_samples += study.allSampleCount\n",
    "    print(study.studyId, study.description, study.allSampleCount)\n",
    "print(\"Total samples:\", total_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_df_to_csv(filename, df):\n",
    "    df.to_csv(filename, index=False)\n",
    "    print(f\"Data written to {filename}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Clinical data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_data_for_study(studyId):\n",
    "\n",
    "    clinical_data_sample = cbioportal.Clinical_Data.getAllClinicalDataInStudyUsingGET(\n",
    "        studyId=studyId,\n",
    "        projection=\"DETAILED\",\n",
    "        clinicalDataType=\"SAMPLE\"\n",
    "    ).result()\n",
    "\n",
    "    clinical_data_patient = cbioportal.Clinical_Data.getAllClinicalDataInStudyUsingGET(\n",
    "        studyId=studyId,\n",
    "        projection=\"DETAILED\",\n",
    "        clinicalDataType=\"PATIENT\"\n",
    "    ).result()\n",
    "\n",
    "    csdf = pd.DataFrame.from_dict([\n",
    "        # python magic that combines two dictionaries:\n",
    "        dict(\n",
    "            {k:getattr(m,k) for k in dir(m)},\n",
    "        )\n",
    "        # create one item in the list for each mutation\n",
    "        for m in clinical_data_sample\n",
    "    ])\n",
    "    cpdf = pd.DataFrame.from_dict([\n",
    "        # python magic that combines two dictionaries:\n",
    "        dict(\n",
    "            {k:getattr(m,k) for k in dir(m)},\n",
    "        )\n",
    "        # create one item in the list for each mutation\n",
    "        for m in clinical_data_patient\n",
    "    ])\n",
    "\n",
    "\n",
    "    cpdf.drop(columns=['clinicalAttribute'], inplace=True)\n",
    "    csdf.drop(columns=['clinicalAttribute'], inplace=True)\n",
    "\n",
    "\n",
    "    mutations = cbioportal.Mutations.getMutationsInMolecularProfileBySampleListIdUsingGET(\n",
    "        molecularProfileId=f'{studyId}_mutations',\n",
    "        sampleListId=f'{studyId}_all',\n",
    "        projection='DETAILED'\n",
    "    ).result()\n",
    "\n",
    "    mdf = pd.DataFrame.from_dict([\n",
    "    # python magic that combines two dictionaries:\n",
    "        dict(\n",
    "            {k:getattr(m,k) for k in dir(m)},\n",
    "            **{k:getattr(m.gene,k) for k in dir(m.gene)}) \n",
    "        # create one item in the list for each mutation\n",
    "        for m in mutations\n",
    "    ])\n",
    "    mdf.drop(columns=['gene', 'uniquePatientKey', 'uniqueSampleKey'], inplace=True)\n",
    "\n",
    "    return cpdf, csdf, mdf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# using these 2 dataframes, i want to create a df with patient id, sample id, age, sex (all clinicalattributeid from cpdf), CANCER_TYPE, (all clinicalattributeid from csdf)\n",
    "def process_patient_sample_data(cpdf, csdf):\n",
    "    # display(cpdf[cpdf.patientId == 'P-CRC-0001'])\n",
    "    # display(csdf[csdf.patientId == 'P-CRC-0001'])\n",
    "    pivot_patient_df = cpdf.pivot(index='patientId', columns='clinicalAttributeId', values='value').reset_index()\n",
    "    pivot_sample_df = csdf.pivot(index=['patientId', 'sampleId', 'studyId'], columns='clinicalAttributeId', values='value').reset_index()\n",
    "    df = pivot_sample_df.merge(pivot_patient_df, on='patientId', how='left')\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def process_study(studyId):\n",
    "    cpdf, csdf, mdf = get_data_for_study(studyId)\n",
    "    df = process_patient_sample_data(cpdf, csdf)\n",
    "    print(df.shape, df.columns)\n",
    "    display(df.head())\n",
    "    print(mdf.shape, mdf.columns)\n",
    "    display(mdf.head())\n",
    "\n",
    "    # drop columsn where all values are null\n",
    "    df.dropna(axis=1, how='all', inplace=True)\n",
    "    mdf.dropna(axis=1, how='all', inplace=True)\n",
    "\n",
    "    write_df_to_csv(f'../../data/cbioportal/{studyId}_clinical_data.csv', df)\n",
    "    write_df_to_csv(f'../../data/cbioportal/{studyId}_mutations.csv', mdf)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def driver():\n",
    "    for study in cll_studies:\n",
    "        studyId = study.studyId\n",
    "        print(studyId)\n",
    "        process_study(studyId)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cllsll_icgc_2011\n",
      "(105, 10) Index(['patientId', 'sampleId', 'studyId', 'CANCER_TYPE',\n",
      "       'CANCER_TYPE_DETAILED', 'MUTATION_COUNT', 'ONCOTREE_CODE',\n",
      "       'SOMATIC_STATUS', 'TMB_NONSYNONYMOUS', 'SAMPLE_COUNT'],\n",
      "      dtype='object', name='clinicalAttributeId')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>clinicalAttributeId</th>\n",
       "      <th>patientId</th>\n",
       "      <th>sampleId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>CANCER_TYPE</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "      <th>MUTATION_COUNT</th>\n",
       "      <th>ONCOTREE_CODE</th>\n",
       "      <th>SOMATIC_STATUS</th>\n",
       "      <th>TMB_NONSYNONYMOUS</th>\n",
       "      <th>SAMPLE_COUNT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CLL_100</td>\n",
       "      <td>CLL_100</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>6</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CLL_110</td>\n",
       "      <td>CLL_110</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>10</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.333333333333</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CLL_117</td>\n",
       "      <td>CLL_117</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>11</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.366666666667</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CLL_124</td>\n",
       "      <td>CLL_124</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>11</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.366666666667</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CLL_13</td>\n",
       "      <td>CLL_13</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>13</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.433333333333</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "clinicalAttributeId patientId sampleId           studyId  \\\n",
       "0                     CLL_100  CLL_100  cllsll_icgc_2011   \n",
       "1                     CLL_110  CLL_110  cllsll_icgc_2011   \n",
       "2                     CLL_117  CLL_117  cllsll_icgc_2011   \n",
       "3                     CLL_124  CLL_124  cllsll_icgc_2011   \n",
       "4                      CLL_13   CLL_13  cllsll_icgc_2011   \n",
       "\n",
       "clinicalAttributeId              CANCER_TYPE  \\\n",
       "0                    Mature B-Cell Neoplasms   \n",
       "1                    Mature B-Cell Neoplasms   \n",
       "2                    Mature B-Cell Neoplasms   \n",
       "3                    Mature B-Cell Neoplasms   \n",
       "4                    Mature B-Cell Neoplasms   \n",
       "\n",
       "clinicalAttributeId                               CANCER_TYPE_DETAILED  \\\n",
       "0                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "1                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "2                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "3                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "4                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "\n",
       "clinicalAttributeId MUTATION_COUNT ONCOTREE_CODE SOMATIC_STATUS  \\\n",
       "0                                6        CLLSLL        Matched   \n",
       "1                               10        CLLSLL        Matched   \n",
       "2                               11        CLLSLL        Matched   \n",
       "3                               11        CLLSLL        Matched   \n",
       "4                               13        CLLSLL        Matched   \n",
       "\n",
       "clinicalAttributeId TMB_NONSYNONYMOUS SAMPLE_COUNT  \n",
       "0                                 0.2            1  \n",
       "1                      0.333333333333            1  \n",
       "2                      0.366666666667            1  \n",
       "3                      0.366666666667            1  \n",
       "4                      0.433333333333            1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1067, 35) Index(['alleleSpecificCopyNumber', 'aminoAcidChange', 'center', 'chr',\n",
      "       'driverFilter', 'driverFilterAnnotation', 'driverTiersFilter',\n",
      "       'driverTiersFilterAnnotation', 'endPosition', 'entrezGeneId', 'keyword',\n",
      "       'molecularProfileId', 'mutationStatus', 'mutationType',\n",
      "       'namespaceColumns', 'ncbiBuild', 'normalAltCount', 'normalRefCount',\n",
      "       'patientId', 'proteinChange', 'proteinPosEnd', 'proteinPosStart',\n",
      "       'referenceAllele', 'refseqMrnaId', 'sampleId', 'startPosition',\n",
      "       'studyId', 'tumorAltCount', 'tumorRefCount', 'validationStatus',\n",
      "       'variantAllele', 'variantType', 'geneticEntityId', 'hugoGeneSymbol',\n",
      "       'type'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alleleSpecificCopyNumber</th>\n",
       "      <th>aminoAcidChange</th>\n",
       "      <th>center</th>\n",
       "      <th>chr</th>\n",
       "      <th>driverFilter</th>\n",
       "      <th>driverFilterAnnotation</th>\n",
       "      <th>driverTiersFilter</th>\n",
       "      <th>driverTiersFilterAnnotation</th>\n",
       "      <th>endPosition</th>\n",
       "      <th>entrezGeneId</th>\n",
       "      <th>...</th>\n",
       "      <th>startPosition</th>\n",
       "      <th>studyId</th>\n",
       "      <th>tumorAltCount</th>\n",
       "      <th>tumorRefCount</th>\n",
       "      <th>validationStatus</th>\n",
       "      <th>variantAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>geneticEntityId</th>\n",
       "      <th>hugoGeneSymbol</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Oviedo</td>\n",
       "      <td>12</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>9021782</td>\n",
       "      <td>144568</td>\n",
       "      <td>...</td>\n",
       "      <td>9021782</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>A2ML1</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Oviedo</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>169391327</td>\n",
       "      <td>57821</td>\n",
       "      <td>...</td>\n",
       "      <td>169391327</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>CCDC181</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Oviedo</td>\n",
       "      <td>14</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>95660226</td>\n",
       "      <td>79789</td>\n",
       "      <td>...</td>\n",
       "      <td>95660226</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>CLMN</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Oviedo</td>\n",
       "      <td>17</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>36002206</td>\n",
       "      <td>11056</td>\n",
       "      <td>...</td>\n",
       "      <td>36002206</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>DDX52</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Oviedo</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>38141882</td>\n",
       "      <td>9940</td>\n",
       "      <td>...</td>\n",
       "      <td>38141882</td>\n",
       "      <td>cllsll_icgc_2011</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>DLEC1</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  alleleSpecificCopyNumber aminoAcidChange  center chr driverFilter  \\\n",
       "0                     None            None  Oviedo  12         None   \n",
       "1                     None            None  Oviedo   1         None   \n",
       "2                     None            None  Oviedo  14         None   \n",
       "3                     None            None  Oviedo  17         None   \n",
       "4                     None            None  Oviedo   3         None   \n",
       "\n",
       "  driverFilterAnnotation driverTiersFilter driverTiersFilterAnnotation  \\\n",
       "0                   None              None                        None   \n",
       "1                   None              None                        None   \n",
       "2                   None              None                        None   \n",
       "3                   None              None                        None   \n",
       "4                   None              None                        None   \n",
       "\n",
       "   endPosition  entrezGeneId  ... startPosition           studyId  \\\n",
       "0      9021782        144568  ...       9021782  cllsll_icgc_2011   \n",
       "1    169391327         57821  ...     169391327  cllsll_icgc_2011   \n",
       "2     95660226         79789  ...      95660226  cllsll_icgc_2011   \n",
       "3     36002206         11056  ...      36002206  cllsll_icgc_2011   \n",
       "4     38141882          9940  ...      38141882  cllsll_icgc_2011   \n",
       "\n",
       "  tumorAltCount tumorRefCount validationStatus variantAllele variantType  \\\n",
       "0          None          None               NA             G         SNP   \n",
       "1          None          None               NA             T         SNP   \n",
       "2          None          None               NA             T         SNP   \n",
       "3          None          None               NA             A         SNP   \n",
       "4          None          None               NA             A         SNP   \n",
       "\n",
       "  geneticEntityId hugoGeneSymbol            type  \n",
       "0            None          A2ML1  protein-coding  \n",
       "1            None        CCDC181  protein-coding  \n",
       "2            None           CLMN  protein-coding  \n",
       "3            None          DDX52  protein-coding  \n",
       "4            None          DLEC1  protein-coding  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data written to ../../data/cllsll_icgc_2011_clinical_data.csv\n",
      "Data written to ../../data/cllsll_icgc_2011_mutations.csv\n",
      "lcll_broad_2013\n",
      "(160, 10) Index(['patientId', 'sampleId', 'studyId', 'CANCER_TYPE',\n",
      "       'CANCER_TYPE_DETAILED', 'MUTATION_COUNT', 'ONCOTREE_CODE',\n",
      "       'SOMATIC_STATUS', 'TMB_NONSYNONYMOUS', 'SAMPLE_COUNT'],\n",
      "      dtype='object', name='clinicalAttributeId')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>clinicalAttributeId</th>\n",
       "      <th>patientId</th>\n",
       "      <th>sampleId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>CANCER_TYPE</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "      <th>MUTATION_COUNT</th>\n",
       "      <th>ONCOTREE_CODE</th>\n",
       "      <th>SOMATIC_STATUS</th>\n",
       "      <th>TMB_NONSYNONYMOUS</th>\n",
       "      <th>SAMPLE_COUNT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CLL001</td>\n",
       "      <td>CLL001</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>13</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.433333333333</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CLL003</td>\n",
       "      <td>CLL003</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>14</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.466666666667</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CLL004</td>\n",
       "      <td>CLL004</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>12</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CLL005</td>\n",
       "      <td>CLL005</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>19</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.633333333333</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CLL006</td>\n",
       "      <td>CLL006</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>29</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.966666666667</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "clinicalAttributeId patientId sampleId          studyId  \\\n",
       "0                      CLL001   CLL001  lcll_broad_2013   \n",
       "1                      CLL003   CLL003  lcll_broad_2013   \n",
       "2                      CLL004   CLL004  lcll_broad_2013   \n",
       "3                      CLL005   CLL005  lcll_broad_2013   \n",
       "4                      CLL006   CLL006  lcll_broad_2013   \n",
       "\n",
       "clinicalAttributeId              CANCER_TYPE  \\\n",
       "0                    Mature B-Cell Neoplasms   \n",
       "1                    Mature B-Cell Neoplasms   \n",
       "2                    Mature B-Cell Neoplasms   \n",
       "3                    Mature B-Cell Neoplasms   \n",
       "4                    Mature B-Cell Neoplasms   \n",
       "\n",
       "clinicalAttributeId                               CANCER_TYPE_DETAILED  \\\n",
       "0                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "1                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "2                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "3                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "4                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "\n",
       "clinicalAttributeId MUTATION_COUNT ONCOTREE_CODE SOMATIC_STATUS  \\\n",
       "0                               13        CLLSLL        Matched   \n",
       "1                               14        CLLSLL        Matched   \n",
       "2                               12        CLLSLL        Matched   \n",
       "3                               19        CLLSLL        Matched   \n",
       "4                               29        CLLSLL        Matched   \n",
       "\n",
       "clinicalAttributeId TMB_NONSYNONYMOUS SAMPLE_COUNT  \n",
       "0                      0.433333333333            1  \n",
       "1                      0.466666666667            1  \n",
       "2                                 0.4            1  \n",
       "3                      0.633333333333            1  \n",
       "4                      0.966666666667            1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2312, 35) Index(['alleleSpecificCopyNumber', 'aminoAcidChange', 'center', 'chr',\n",
      "       'driverFilter', 'driverFilterAnnotation', 'driverTiersFilter',\n",
      "       'driverTiersFilterAnnotation', 'endPosition', 'entrezGeneId', 'keyword',\n",
      "       'molecularProfileId', 'mutationStatus', 'mutationType',\n",
      "       'namespaceColumns', 'ncbiBuild', 'normalAltCount', 'normalRefCount',\n",
      "       'patientId', 'proteinChange', 'proteinPosEnd', 'proteinPosStart',\n",
      "       'referenceAllele', 'refseqMrnaId', 'sampleId', 'startPosition',\n",
      "       'studyId', 'tumorAltCount', 'tumorRefCount', 'validationStatus',\n",
      "       'variantAllele', 'variantType', 'geneticEntityId', 'hugoGeneSymbol',\n",
      "       'type'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alleleSpecificCopyNumber</th>\n",
       "      <th>aminoAcidChange</th>\n",
       "      <th>center</th>\n",
       "      <th>chr</th>\n",
       "      <th>driverFilter</th>\n",
       "      <th>driverFilterAnnotation</th>\n",
       "      <th>driverTiersFilter</th>\n",
       "      <th>driverTiersFilterAnnotation</th>\n",
       "      <th>endPosition</th>\n",
       "      <th>entrezGeneId</th>\n",
       "      <th>...</th>\n",
       "      <th>startPosition</th>\n",
       "      <th>studyId</th>\n",
       "      <th>tumorAltCount</th>\n",
       "      <th>tumorRefCount</th>\n",
       "      <th>validationStatus</th>\n",
       "      <th>variantAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>geneticEntityId</th>\n",
       "      <th>hugoGeneSymbol</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>38182641</td>\n",
       "      <td>4615</td>\n",
       "      <td>...</td>\n",
       "      <td>38182641</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>MYD88</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>12</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>86373542</td>\n",
       "      <td>25834</td>\n",
       "      <td>...</td>\n",
       "      <td>86373542</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>MGAT4C</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>12</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>76741541</td>\n",
       "      <td>79738</td>\n",
       "      <td>...</td>\n",
       "      <td>76741541</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>BBS10</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>56419944</td>\n",
       "      <td>114800</td>\n",
       "      <td>...</td>\n",
       "      <td>56419944</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>CCDC85A</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>52812956</td>\n",
       "      <td>3697</td>\n",
       "      <td>...</td>\n",
       "      <td>52812956</td>\n",
       "      <td>lcll_broad_2013</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>ITIH1</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  alleleSpecificCopyNumber aminoAcidChange         center chr driverFilter  \\\n",
       "0                     None            None  broad.mit.edu   3         None   \n",
       "1                     None            None  broad.mit.edu  12         None   \n",
       "2                     None            None  broad.mit.edu  12         None   \n",
       "3                     None            None  broad.mit.edu   2         None   \n",
       "4                     None            None  broad.mit.edu   3         None   \n",
       "\n",
       "  driverFilterAnnotation driverTiersFilter driverTiersFilterAnnotation  \\\n",
       "0                   None              None                        None   \n",
       "1                   None              None                        None   \n",
       "2                   None              None                        None   \n",
       "3                   None              None                        None   \n",
       "4                   None              None                        None   \n",
       "\n",
       "   endPosition  entrezGeneId  ... startPosition          studyId  \\\n",
       "0     38182641          4615  ...      38182641  lcll_broad_2013   \n",
       "1     86373542         25834  ...      86373542  lcll_broad_2013   \n",
       "2     76741541         79738  ...      76741541  lcll_broad_2013   \n",
       "3     56419944        114800  ...      56419944  lcll_broad_2013   \n",
       "4     52812956          3697  ...      52812956  lcll_broad_2013   \n",
       "\n",
       "  tumorAltCount tumorRefCount validationStatus variantAllele variantType  \\\n",
       "0          None          None               NA             C         SNP   \n",
       "1          None          None               NA             A         SNP   \n",
       "2          None          None               NA             C         SNP   \n",
       "3          None          None               NA             G         SNP   \n",
       "4          None          None               NA             T         SNP   \n",
       "\n",
       "  geneticEntityId hugoGeneSymbol            type  \n",
       "0            None          MYD88  protein-coding  \n",
       "1            None         MGAT4C  protein-coding  \n",
       "2            None          BBS10  protein-coding  \n",
       "3            None        CCDC85A  protein-coding  \n",
       "4            None          ITIH1  protein-coding  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data written to ../../data/lcll_broad_2013_clinical_data.csv\n",
      "Data written to ../../data/lcll_broad_2013_mutations.csv\n",
      "cll_iuopa_2015\n",
      "(506, 16) Index(['patientId', 'sampleId', 'studyId', 'CANCER_TYPE',\n",
      "       'CANCER_TYPE_DETAILED', 'EXOME_CAPTURE', 'IGHV_GENE', 'IGHV_IDENTITY',\n",
      "       'IGHV_MUTATED', 'MUTATION_COUNT', 'ONCOTREE_CODE', 'PLATFORM',\n",
      "       'SOMATIC_STATUS', 'TMB_NONSYNONYMOUS', 'EPIGENETIC', 'SAMPLE_COUNT'],\n",
      "      dtype='object', name='clinicalAttributeId')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>clinicalAttributeId</th>\n",
       "      <th>patientId</th>\n",
       "      <th>sampleId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>CANCER_TYPE</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "      <th>EXOME_CAPTURE</th>\n",
       "      <th>IGHV_GENE</th>\n",
       "      <th>IGHV_IDENTITY</th>\n",
       "      <th>IGHV_MUTATED</th>\n",
       "      <th>MUTATION_COUNT</th>\n",
       "      <th>ONCOTREE_CODE</th>\n",
       "      <th>PLATFORM</th>\n",
       "      <th>SOMATIC_STATUS</th>\n",
       "      <th>TMB_NONSYNONYMOUS</th>\n",
       "      <th>EPIGENETIC</th>\n",
       "      <th>SAMPLE_COUNT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cll_iuopa_2015_1</td>\n",
       "      <td>cll_iuopa_2015_1</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>SureSelect Human All Exon</td>\n",
       "      <td>IGHV1-3</td>\n",
       "      <td>100</td>\n",
       "      <td>UNMUT</td>\n",
       "      <td>12</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>WES,WGS</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.4</td>\n",
       "      <td>Naive-like CLL</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cll_iuopa_2015_10</td>\n",
       "      <td>cll_iuopa_2015_10</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>SureSelect XT Human All Exon 50 Mb</td>\n",
       "      <td>IGHV3-21</td>\n",
       "      <td>99.6</td>\n",
       "      <td>UNMUT</td>\n",
       "      <td>23</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>WES,WGS</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.766666666667</td>\n",
       "      <td>Intermediate CLL</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cll_iuopa_2015_100</td>\n",
       "      <td>cll_iuopa_2015_100</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>SureSelect XT Human All Exon 50 Mb</td>\n",
       "      <td>IGHV1-69</td>\n",
       "      <td>100</td>\n",
       "      <td>UNMUT</td>\n",
       "      <td>7</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>WES</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.233333333333</td>\n",
       "      <td>Naive-like CLL</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cll_iuopa_2015_1047</td>\n",
       "      <td>cll_iuopa_2015_1047</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>SureSelect XT2 Human All Exon v4+UTRs</td>\n",
       "      <td>IGHV4-59</td>\n",
       "      <td>97.19</td>\n",
       "      <td>MUT</td>\n",
       "      <td>8</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>WES</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.266666666667</td>\n",
       "      <td>Memory-like CLL</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cll_iuopa_2015_1050</td>\n",
       "      <td>cll_iuopa_2015_1050</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>SureSelect XT2 Human All Exon v4+UTRs</td>\n",
       "      <td>IGHV1-69</td>\n",
       "      <td>100</td>\n",
       "      <td>UNMUT</td>\n",
       "      <td>13</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>WES</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.433333333333</td>\n",
       "      <td>Naive-like CLL</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "clinicalAttributeId            patientId             sampleId         studyId  \\\n",
       "0                       cll_iuopa_2015_1     cll_iuopa_2015_1  cll_iuopa_2015   \n",
       "1                      cll_iuopa_2015_10    cll_iuopa_2015_10  cll_iuopa_2015   \n",
       "2                     cll_iuopa_2015_100   cll_iuopa_2015_100  cll_iuopa_2015   \n",
       "3                    cll_iuopa_2015_1047  cll_iuopa_2015_1047  cll_iuopa_2015   \n",
       "4                    cll_iuopa_2015_1050  cll_iuopa_2015_1050  cll_iuopa_2015   \n",
       "\n",
       "clinicalAttributeId              CANCER_TYPE  \\\n",
       "0                    Mature B-Cell Neoplasms   \n",
       "1                    Mature B-Cell Neoplasms   \n",
       "2                    Mature B-Cell Neoplasms   \n",
       "3                    Mature B-Cell Neoplasms   \n",
       "4                    Mature B-Cell Neoplasms   \n",
       "\n",
       "clinicalAttributeId                               CANCER_TYPE_DETAILED  \\\n",
       "0                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "1                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "2                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "3                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "4                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "\n",
       "clinicalAttributeId                          EXOME_CAPTURE IGHV_GENE  \\\n",
       "0                                SureSelect Human All Exon   IGHV1-3   \n",
       "1                       SureSelect XT Human All Exon 50 Mb  IGHV3-21   \n",
       "2                       SureSelect XT Human All Exon 50 Mb  IGHV1-69   \n",
       "3                    SureSelect XT2 Human All Exon v4+UTRs  IGHV4-59   \n",
       "4                    SureSelect XT2 Human All Exon v4+UTRs  IGHV1-69   \n",
       "\n",
       "clinicalAttributeId IGHV_IDENTITY IGHV_MUTATED MUTATION_COUNT ONCOTREE_CODE  \\\n",
       "0                             100        UNMUT             12        CLLSLL   \n",
       "1                            99.6        UNMUT             23        CLLSLL   \n",
       "2                             100        UNMUT              7        CLLSLL   \n",
       "3                           97.19          MUT              8        CLLSLL   \n",
       "4                             100        UNMUT             13        CLLSLL   \n",
       "\n",
       "clinicalAttributeId PLATFORM SOMATIC_STATUS TMB_NONSYNONYMOUS  \\\n",
       "0                    WES,WGS        Matched               0.4   \n",
       "1                    WES,WGS        Matched    0.766666666667   \n",
       "2                        WES        Matched    0.233333333333   \n",
       "3                        WES        Matched    0.266666666667   \n",
       "4                        WES        Matched    0.433333333333   \n",
       "\n",
       "clinicalAttributeId        EPIGENETIC SAMPLE_COUNT  \n",
       "0                      Naive-like CLL            1  \n",
       "1                    Intermediate CLL            1  \n",
       "2                      Naive-like CLL            1  \n",
       "3                     Memory-like CLL            1  \n",
       "4                      Naive-like CLL            1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7134, 35) Index(['alleleSpecificCopyNumber', 'aminoAcidChange', 'center', 'chr',\n",
      "       'driverFilter', 'driverFilterAnnotation', 'driverTiersFilter',\n",
      "       'driverTiersFilterAnnotation', 'endPosition', 'entrezGeneId', 'keyword',\n",
      "       'molecularProfileId', 'mutationStatus', 'mutationType',\n",
      "       'namespaceColumns', 'ncbiBuild', 'normalAltCount', 'normalRefCount',\n",
      "       'patientId', 'proteinChange', 'proteinPosEnd', 'proteinPosStart',\n",
      "       'referenceAllele', 'refseqMrnaId', 'sampleId', 'startPosition',\n",
      "       'studyId', 'tumorAltCount', 'tumorRefCount', 'validationStatus',\n",
      "       'variantAllele', 'variantType', 'geneticEntityId', 'hugoGeneSymbol',\n",
      "       'type'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alleleSpecificCopyNumber</th>\n",
       "      <th>aminoAcidChange</th>\n",
       "      <th>center</th>\n",
       "      <th>chr</th>\n",
       "      <th>driverFilter</th>\n",
       "      <th>driverFilterAnnotation</th>\n",
       "      <th>driverTiersFilter</th>\n",
       "      <th>driverTiersFilterAnnotation</th>\n",
       "      <th>endPosition</th>\n",
       "      <th>entrezGeneId</th>\n",
       "      <th>...</th>\n",
       "      <th>startPosition</th>\n",
       "      <th>studyId</th>\n",
       "      <th>tumorAltCount</th>\n",
       "      <th>tumorRefCount</th>\n",
       "      <th>validationStatus</th>\n",
       "      <th>variantAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>geneticEntityId</th>\n",
       "      <th>hugoGeneSymbol</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>www.unioviedo.es/IUOPA/</td>\n",
       "      <td>19</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>54646887</td>\n",
       "      <td>4849</td>\n",
       "      <td>...</td>\n",
       "      <td>54646887</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>CNOT3</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>www.unioviedo.es/IUOPA/</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>78700896</td>\n",
       "      <td>6091</td>\n",
       "      <td>...</td>\n",
       "      <td>78700896</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>ROBO1</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>www.unioviedo.es/IUOPA/</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>20778606</td>\n",
       "      <td>340273</td>\n",
       "      <td>...</td>\n",
       "      <td>20778606</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>ABCB5</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>www.unioviedo.es/IUOPA/</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>71665885</td>\n",
       "      <td>135152</td>\n",
       "      <td>...</td>\n",
       "      <td>71665885</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>B3GAT2</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>www.unioviedo.es/IUOPA/</td>\n",
       "      <td>19</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>41383838</td>\n",
       "      <td>1549</td>\n",
       "      <td>...</td>\n",
       "      <td>41383838</td>\n",
       "      <td>cll_iuopa_2015</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>CYP2A7</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  alleleSpecificCopyNumber aminoAcidChange                   center chr  \\\n",
       "0                     None            None  www.unioviedo.es/IUOPA/  19   \n",
       "1                     None            None  www.unioviedo.es/IUOPA/   3   \n",
       "2                     None            None  www.unioviedo.es/IUOPA/   7   \n",
       "3                     None            None  www.unioviedo.es/IUOPA/   6   \n",
       "4                     None            None  www.unioviedo.es/IUOPA/  19   \n",
       "\n",
       "  driverFilter driverFilterAnnotation driverTiersFilter  \\\n",
       "0         None                   None              None   \n",
       "1         None                   None              None   \n",
       "2         None                   None              None   \n",
       "3         None                   None              None   \n",
       "4         None                   None              None   \n",
       "\n",
       "  driverTiersFilterAnnotation  endPosition  entrezGeneId  ... startPosition  \\\n",
       "0                        None     54646887          4849  ...      54646887   \n",
       "1                        None     78700896          6091  ...      78700896   \n",
       "2                        None     20778606        340273  ...      20778606   \n",
       "3                        None     71665885        135152  ...      71665885   \n",
       "4                        None     41383838          1549  ...      41383838   \n",
       "\n",
       "          studyId tumorAltCount tumorRefCount validationStatus variantAllele  \\\n",
       "0  cll_iuopa_2015          None          None               NA             A   \n",
       "1  cll_iuopa_2015          None          None               NA             A   \n",
       "2  cll_iuopa_2015          None          None               NA             T   \n",
       "3  cll_iuopa_2015          None          None               NA             A   \n",
       "4  cll_iuopa_2015          None          None               NA             A   \n",
       "\n",
       "  variantType geneticEntityId hugoGeneSymbol            type  \n",
       "0         SNP            None          CNOT3  protein-coding  \n",
       "1         SNP            None          ROBO1  protein-coding  \n",
       "2         SNP            None          ABCB5  protein-coding  \n",
       "3         SNP            None         B3GAT2  protein-coding  \n",
       "4         SNP            None         CYP2A7  protein-coding  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data written to ../../data/cll_iuopa_2015_clinical_data.csv\n",
      "Data written to ../../data/cll_iuopa_2015_mutations.csv\n",
      "cll_broad_2015\n",
      "(537, 14) Index(['patientId', 'sampleId', 'studyId', 'CANCER_TYPE',\n",
      "       'CANCER_TYPE_DETAILED', 'CAPTURE_BAIT_SET', 'IGHV_GENE',\n",
      "       'MUTATION_COUNT', 'ONCOTREE_CODE', 'SOMATIC_STATUS',\n",
      "       'TMB_NONSYNONYMOUS', 'COHORT', 'PRIOR_TREATMENT', 'SAMPLE_COUNT'],\n",
      "      dtype='object', name='clinicalAttributeId')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>clinicalAttributeId</th>\n",
       "      <th>patientId</th>\n",
       "      <th>sampleId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>CANCER_TYPE</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "      <th>CAPTURE_BAIT_SET</th>\n",
       "      <th>IGHV_GENE</th>\n",
       "      <th>MUTATION_COUNT</th>\n",
       "      <th>ONCOTREE_CODE</th>\n",
       "      <th>SOMATIC_STATUS</th>\n",
       "      <th>TMB_NONSYNONYMOUS</th>\n",
       "      <th>COHORT</th>\n",
       "      <th>PRIOR_TREATMENT</th>\n",
       "      <th>SAMPLE_COUNT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CLL-GCLL-0001</td>\n",
       "      <td>CLL-GCLL-0001-Tumor-SM-41JLZ</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>whole_exome_agilent_1.1_refseq_plus_3_boosters</td>\n",
       "      <td>mutated</td>\n",
       "      <td>8</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.266666666667</td>\n",
       "      <td>GCLLSG_CLL8</td>\n",
       "      <td>treatment naive</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CLL-GCLL-0002</td>\n",
       "      <td>CLL-GCLL-0002-Tumor-SM-41JM1</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>whole_exome_agilent_1.1_refseq_plus_3_boosters</td>\n",
       "      <td>mutated</td>\n",
       "      <td>12</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.4</td>\n",
       "      <td>GCLLSG_CLL8</td>\n",
       "      <td>treatment naive</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CLL-GCLL-0003</td>\n",
       "      <td>CLL-GCLL-0003-Tumor-SM-41JM2</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>whole_exome_agilent_1.1_refseq_plus_3_boosters</td>\n",
       "      <td>mutated</td>\n",
       "      <td>14</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.466666666667</td>\n",
       "      <td>GCLLSG_CLL8</td>\n",
       "      <td>treatment naive</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CLL-GCLL-0004</td>\n",
       "      <td>CLL-GCLL-0004-Tumor-SM-41JM3</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>whole_exome_agilent_1.1_refseq_plus_3_boosters</td>\n",
       "      <td>mutated</td>\n",
       "      <td>9</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.3</td>\n",
       "      <td>GCLLSG_CLL8</td>\n",
       "      <td>treatment naive</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CLL-GCLL-0006</td>\n",
       "      <td>CLL-GCLL-0006-Tumor-SM-41JM5</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>whole_exome_agilent_1.1_refseq_plus_3_boosters</td>\n",
       "      <td>mutated</td>\n",
       "      <td>13</td>\n",
       "      <td>CLLSLL</td>\n",
       "      <td>Matched</td>\n",
       "      <td>0.4</td>\n",
       "      <td>GCLLSG_CLL8</td>\n",
       "      <td>treatment naive</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "clinicalAttributeId      patientId                      sampleId  \\\n",
       "0                    CLL-GCLL-0001  CLL-GCLL-0001-Tumor-SM-41JLZ   \n",
       "1                    CLL-GCLL-0002  CLL-GCLL-0002-Tumor-SM-41JM1   \n",
       "2                    CLL-GCLL-0003  CLL-GCLL-0003-Tumor-SM-41JM2   \n",
       "3                    CLL-GCLL-0004  CLL-GCLL-0004-Tumor-SM-41JM3   \n",
       "4                    CLL-GCLL-0006  CLL-GCLL-0006-Tumor-SM-41JM5   \n",
       "\n",
       "clinicalAttributeId         studyId              CANCER_TYPE  \\\n",
       "0                    cll_broad_2015  Mature B-Cell Neoplasms   \n",
       "1                    cll_broad_2015  Mature B-Cell Neoplasms   \n",
       "2                    cll_broad_2015  Mature B-Cell Neoplasms   \n",
       "3                    cll_broad_2015  Mature B-Cell Neoplasms   \n",
       "4                    cll_broad_2015  Mature B-Cell Neoplasms   \n",
       "\n",
       "clinicalAttributeId                               CANCER_TYPE_DETAILED  \\\n",
       "0                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "1                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "2                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "3                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "4                    Chronic Lymphocytic Leukemia/Small Lymphocytic...   \n",
       "\n",
       "clinicalAttributeId                                CAPTURE_BAIT_SET IGHV_GENE  \\\n",
       "0                    whole_exome_agilent_1.1_refseq_plus_3_boosters   mutated   \n",
       "1                    whole_exome_agilent_1.1_refseq_plus_3_boosters   mutated   \n",
       "2                    whole_exome_agilent_1.1_refseq_plus_3_boosters   mutated   \n",
       "3                    whole_exome_agilent_1.1_refseq_plus_3_boosters   mutated   \n",
       "4                    whole_exome_agilent_1.1_refseq_plus_3_boosters   mutated   \n",
       "\n",
       "clinicalAttributeId MUTATION_COUNT ONCOTREE_CODE SOMATIC_STATUS  \\\n",
       "0                                8        CLLSLL        Matched   \n",
       "1                               12        CLLSLL        Matched   \n",
       "2                               14        CLLSLL        Matched   \n",
       "3                                9        CLLSLL        Matched   \n",
       "4                               13        CLLSLL        Matched   \n",
       "\n",
       "clinicalAttributeId TMB_NONSYNONYMOUS       COHORT  PRIOR_TREATMENT  \\\n",
       "0                      0.266666666667  GCLLSG_CLL8  treatment naive   \n",
       "1                                 0.4  GCLLSG_CLL8  treatment naive   \n",
       "2                      0.466666666667  GCLLSG_CLL8  treatment naive   \n",
       "3                                 0.3  GCLLSG_CLL8  treatment naive   \n",
       "4                                 0.4  GCLLSG_CLL8  treatment naive   \n",
       "\n",
       "clinicalAttributeId SAMPLE_COUNT  \n",
       "0                              1  \n",
       "1                              1  \n",
       "2                              1  \n",
       "3                              1  \n",
       "4                              1  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(9487, 35) Index(['alleleSpecificCopyNumber', 'aminoAcidChange', 'center', 'chr',\n",
      "       'driverFilter', 'driverFilterAnnotation', 'driverTiersFilter',\n",
      "       'driverTiersFilterAnnotation', 'endPosition', 'entrezGeneId', 'keyword',\n",
      "       'molecularProfileId', 'mutationStatus', 'mutationType',\n",
      "       'namespaceColumns', 'ncbiBuild', 'normalAltCount', 'normalRefCount',\n",
      "       'patientId', 'proteinChange', 'proteinPosEnd', 'proteinPosStart',\n",
      "       'referenceAllele', 'refseqMrnaId', 'sampleId', 'startPosition',\n",
      "       'studyId', 'tumorAltCount', 'tumorRefCount', 'validationStatus',\n",
      "       'variantAllele', 'variantType', 'geneticEntityId', 'hugoGeneSymbol',\n",
      "       'type'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alleleSpecificCopyNumber</th>\n",
       "      <th>aminoAcidChange</th>\n",
       "      <th>center</th>\n",
       "      <th>chr</th>\n",
       "      <th>driverFilter</th>\n",
       "      <th>driverFilterAnnotation</th>\n",
       "      <th>driverTiersFilter</th>\n",
       "      <th>driverTiersFilterAnnotation</th>\n",
       "      <th>endPosition</th>\n",
       "      <th>entrezGeneId</th>\n",
       "      <th>...</th>\n",
       "      <th>startPosition</th>\n",
       "      <th>studyId</th>\n",
       "      <th>tumorAltCount</th>\n",
       "      <th>tumorRefCount</th>\n",
       "      <th>validationStatus</th>\n",
       "      <th>variantAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>geneticEntityId</th>\n",
       "      <th>hugoGeneSymbol</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>129802568</td>\n",
       "      <td>3908</td>\n",
       "      <td>...</td>\n",
       "      <td>129802568</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>63</td>\n",
       "      <td>97</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>LAMA2</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>120438518</td>\n",
       "      <td>11085</td>\n",
       "      <td>...</td>\n",
       "      <td>120438518</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>4</td>\n",
       "      <td>116</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>ADAM30</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>20</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4229250</td>\n",
       "      <td>146</td>\n",
       "      <td>...</td>\n",
       "      <td>4229250</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>14</td>\n",
       "      <td>24</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>ADRA1D</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>196799323</td>\n",
       "      <td>56171</td>\n",
       "      <td>...</td>\n",
       "      <td>196799323</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>78</td>\n",
       "      <td>105</td>\n",
       "      <td>NA</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>DNAH7</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>broad.mit.edu</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>228612633</td>\n",
       "      <td>8290</td>\n",
       "      <td>...</td>\n",
       "      <td>228612633</td>\n",
       "      <td>cll_broad_2015</td>\n",
       "      <td>43</td>\n",
       "      <td>34</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>H3-4</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  alleleSpecificCopyNumber aminoAcidChange         center chr driverFilter  \\\n",
       "0                     None            None  broad.mit.edu   6         None   \n",
       "1                     None            None  broad.mit.edu   1         None   \n",
       "2                     None            None  broad.mit.edu  20         None   \n",
       "3                     None            None  broad.mit.edu   2         None   \n",
       "4                     None            None  broad.mit.edu   1         None   \n",
       "\n",
       "  driverFilterAnnotation driverTiersFilter driverTiersFilterAnnotation  \\\n",
       "0                   None              None                        None   \n",
       "1                   None              None                        None   \n",
       "2                   None              None                        None   \n",
       "3                   None              None                        None   \n",
       "4                   None              None                        None   \n",
       "\n",
       "   endPosition  entrezGeneId  ... startPosition         studyId tumorAltCount  \\\n",
       "0    129802568          3908  ...     129802568  cll_broad_2015            63   \n",
       "1    120438518         11085  ...     120438518  cll_broad_2015             4   \n",
       "2      4229250           146  ...       4229250  cll_broad_2015            14   \n",
       "3    196799323         56171  ...     196799323  cll_broad_2015            78   \n",
       "4    228612633          8290  ...     228612633  cll_broad_2015            43   \n",
       "\n",
       "  tumorRefCount validationStatus variantAllele variantType geneticEntityId  \\\n",
       "0            97               NA             A         SNP            None   \n",
       "1           116               NA             A         SNP            None   \n",
       "2            24               NA             A         SNP            None   \n",
       "3           105               NA             C         SNP            None   \n",
       "4            34               NA             A         SNP            None   \n",
       "\n",
       "  hugoGeneSymbol            type  \n",
       "0          LAMA2  protein-coding  \n",
       "1         ADAM30  protein-coding  \n",
       "2         ADRA1D  protein-coding  \n",
       "3          DNAH7  protein-coding  \n",
       "4           H3-4  protein-coding  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data written to ../../data/cll_broad_2015_clinical_data.csv\n",
      "Data written to ../../data/cll_broad_2015_mutations.csv\n",
      "cll_broad_2022\n",
      "(1154, 43) Index(['patientId', 'sampleId', 'studyId', 'ARRAY450K', 'CANCER_TYPE',\n",
      "       'CANCER_TYPE_DETAILED', 'CENTER', 'CLL_EPITYPE', 'DISEASE_TYPE',\n",
      "       'EC_DISCOVERY', 'EC_EXTENSION', 'IGLV3_21_R110', 'M-CLL_WES',\n",
      "       'MUTATION_COUNT', 'NORMAL_MEAN_COVERAGE', 'ONCOTREE_CODE',\n",
      "       'RNA_SEQUENCING', 'RRBS', 'SEQUENCING_PLATFORM', 'SOMATIC_STATUS',\n",
      "       'TMB_NONSYNONYMOUS', 'TUMOR_MEAN_COVERAGE', 'TUMOR_MOLECULAR_SUBTYPE',\n",
      "       'TUMOR_SAMPLE_PLOIDY', 'TUMOR_SAMPLE_PURITY', 'U-CLL_WES', 'U1_STATUS',\n",
      "       'AGE', 'AGE_SAMPLING', 'COHORT', 'DEATH_DAYS', 'EXPRESSION_CLUSTER',\n",
      "       'FFS_MONTHS', 'FFS_STATUS', 'IGHV_IDENTITY_PERCENTAGE',\n",
      "       'IGHV_MUTATION_STATUS', 'OS_MONTHS', 'OS_STATUS',\n",
      "       'PRIOR_TREATMENT_CATEGORY', 'SAMPLE_COUNT', 'SEX',\n",
      "       'TREATMENT_AFTER_SAMPLING', 'TREATMENT_STATUS'],\n",
      "      dtype='object', name='clinicalAttributeId')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>clinicalAttributeId</th>\n",
       "      <th>patientId</th>\n",
       "      <th>sampleId</th>\n",
       "      <th>studyId</th>\n",
       "      <th>ARRAY450K</th>\n",
       "      <th>CANCER_TYPE</th>\n",
       "      <th>CANCER_TYPE_DETAILED</th>\n",
       "      <th>CENTER</th>\n",
       "      <th>CLL_EPITYPE</th>\n",
       "      <th>DISEASE_TYPE</th>\n",
       "      <th>EC_DISCOVERY</th>\n",
       "      <th>...</th>\n",
       "      <th>FFS_STATUS</th>\n",
       "      <th>IGHV_IDENTITY_PERCENTAGE</th>\n",
       "      <th>IGHV_MUTATION_STATUS</th>\n",
       "      <th>OS_MONTHS</th>\n",
       "      <th>OS_STATUS</th>\n",
       "      <th>PRIOR_TREATMENT_CATEGORY</th>\n",
       "      <th>SAMPLE_COUNT</th>\n",
       "      <th>SEX</th>\n",
       "      <th>TREATMENT_AFTER_SAMPLING</th>\n",
       "      <th>TREATMENT_STATUS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-CRC-0001</td>\n",
       "      <td>CRC-0001</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>No</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>Broad</td>\n",
       "      <td>n-CLL</td>\n",
       "      <td>CLL</td>\n",
       "      <td>Yes</td>\n",
       "      <td>...</td>\n",
       "      <td>1:Failure</td>\n",
       "      <td>100</td>\n",
       "      <td>unmutated</td>\n",
       "      <td>147.19</td>\n",
       "      <td>1:DECEASED</td>\n",
       "      <td>Untreated</td>\n",
       "      <td>1</td>\n",
       "      <td>Female</td>\n",
       "      <td>Chemo + Ab</td>\n",
       "      <td>Pre-treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-CRC-0002</td>\n",
       "      <td>CRC-0002</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>No</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>Broad</td>\n",
       "      <td>n-CLL</td>\n",
       "      <td>CLL</td>\n",
       "      <td>Yes</td>\n",
       "      <td>...</td>\n",
       "      <td>1:Failure</td>\n",
       "      <td>100</td>\n",
       "      <td>unmutated</td>\n",
       "      <td>154.49</td>\n",
       "      <td>1:DECEASED</td>\n",
       "      <td>Untreated</td>\n",
       "      <td>1</td>\n",
       "      <td>Male</td>\n",
       "      <td>Chemo + Ab</td>\n",
       "      <td>Pre-treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-CRC-0003</td>\n",
       "      <td>CRC-0003</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>No</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>Broad</td>\n",
       "      <td>n-CLL</td>\n",
       "      <td>CLL</td>\n",
       "      <td>Yes</td>\n",
       "      <td>...</td>\n",
       "      <td>1:Failure</td>\n",
       "      <td>100</td>\n",
       "      <td>unmutated</td>\n",
       "      <td>51.25</td>\n",
       "      <td>1:DECEASED</td>\n",
       "      <td>Untreated</td>\n",
       "      <td>1</td>\n",
       "      <td>Female</td>\n",
       "      <td>Chemo + Ab</td>\n",
       "      <td>Pre-treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-CRC-0004</td>\n",
       "      <td>CRC-0004</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>No</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>Broad</td>\n",
       "      <td>m-CLL</td>\n",
       "      <td>CLL</td>\n",
       "      <td>Yes</td>\n",
       "      <td>...</td>\n",
       "      <td>1:Failure</td>\n",
       "      <td>92.28</td>\n",
       "      <td>mutated</td>\n",
       "      <td>183.55</td>\n",
       "      <td>0:LIVING</td>\n",
       "      <td>Untreated</td>\n",
       "      <td>1</td>\n",
       "      <td>Male</td>\n",
       "      <td>Chemo + Ab</td>\n",
       "      <td>Pre-treatment</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-CRC-0005</td>\n",
       "      <td>CRC-0005</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>No</td>\n",
       "      <td>Mature B-Cell Neoplasms</td>\n",
       "      <td>Chronic Lymphocytic Leukemia/Small Lymphocytic...</td>\n",
       "      <td>Broad</td>\n",
       "      <td>n-CLL</td>\n",
       "      <td>CLL</td>\n",
       "      <td>Yes</td>\n",
       "      <td>...</td>\n",
       "      <td>1:Failure</td>\n",
       "      <td>100</td>\n",
       "      <td>unmutated</td>\n",
       "      <td>164.35</td>\n",
       "      <td>0:LIVING</td>\n",
       "      <td>Untreated</td>\n",
       "      <td>1</td>\n",
       "      <td>Male</td>\n",
       "      <td>Chemo + Ab</td>\n",
       "      <td>Pre-treatment</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "clinicalAttributeId   patientId  sampleId         studyId ARRAY450K  \\\n",
       "0                    P-CRC-0001  CRC-0001  cll_broad_2022        No   \n",
       "1                    P-CRC-0002  CRC-0002  cll_broad_2022        No   \n",
       "2                    P-CRC-0003  CRC-0003  cll_broad_2022        No   \n",
       "3                    P-CRC-0004  CRC-0004  cll_broad_2022        No   \n",
       "4                    P-CRC-0005  CRC-0005  cll_broad_2022        No   \n",
       "\n",
       "clinicalAttributeId              CANCER_TYPE  \\\n",
       "0                    Mature B-Cell Neoplasms   \n",
       "1                    Mature B-Cell Neoplasms   \n",
       "2                    Mature B-Cell Neoplasms   \n",
       "3                    Mature B-Cell Neoplasms   \n",
       "4                    Mature B-Cell Neoplasms   \n",
       "\n",
       "clinicalAttributeId                               CANCER_TYPE_DETAILED CENTER  \\\n",
       "0                    Chronic Lymphocytic Leukemia/Small Lymphocytic...  Broad   \n",
       "1                    Chronic Lymphocytic Leukemia/Small Lymphocytic...  Broad   \n",
       "2                    Chronic Lymphocytic Leukemia/Small Lymphocytic...  Broad   \n",
       "3                    Chronic Lymphocytic Leukemia/Small Lymphocytic...  Broad   \n",
       "4                    Chronic Lymphocytic Leukemia/Small Lymphocytic...  Broad   \n",
       "\n",
       "clinicalAttributeId CLL_EPITYPE DISEASE_TYPE EC_DISCOVERY  ... FFS_STATUS  \\\n",
       "0                         n-CLL          CLL          Yes  ...  1:Failure   \n",
       "1                         n-CLL          CLL          Yes  ...  1:Failure   \n",
       "2                         n-CLL          CLL          Yes  ...  1:Failure   \n",
       "3                         m-CLL          CLL          Yes  ...  1:Failure   \n",
       "4                         n-CLL          CLL          Yes  ...  1:Failure   \n",
       "\n",
       "clinicalAttributeId IGHV_IDENTITY_PERCENTAGE IGHV_MUTATION_STATUS OS_MONTHS  \\\n",
       "0                                        100            unmutated    147.19   \n",
       "1                                        100            unmutated    154.49   \n",
       "2                                        100            unmutated     51.25   \n",
       "3                                      92.28              mutated    183.55   \n",
       "4                                        100            unmutated    164.35   \n",
       "\n",
       "clinicalAttributeId   OS_STATUS PRIOR_TREATMENT_CATEGORY SAMPLE_COUNT     SEX  \\\n",
       "0                    1:DECEASED                Untreated            1  Female   \n",
       "1                    1:DECEASED                Untreated            1    Male   \n",
       "2                    1:DECEASED                Untreated            1  Female   \n",
       "3                      0:LIVING                Untreated            1    Male   \n",
       "4                      0:LIVING                Untreated            1    Male   \n",
       "\n",
       "clinicalAttributeId TREATMENT_AFTER_SAMPLING TREATMENT_STATUS  \n",
       "0                                 Chemo + Ab    Pre-treatment  \n",
       "1                                 Chemo + Ab    Pre-treatment  \n",
       "2                                 Chemo + Ab    Pre-treatment  \n",
       "3                                 Chemo + Ab    Pre-treatment  \n",
       "4                                 Chemo + Ab    Pre-treatment  \n",
       "\n",
       "[5 rows x 43 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(30229, 35) Index(['alleleSpecificCopyNumber', 'aminoAcidChange', 'center', 'chr',\n",
      "       'driverFilter', 'driverFilterAnnotation', 'driverTiersFilter',\n",
      "       'driverTiersFilterAnnotation', 'endPosition', 'entrezGeneId', 'keyword',\n",
      "       'molecularProfileId', 'mutationStatus', 'mutationType',\n",
      "       'namespaceColumns', 'ncbiBuild', 'normalAltCount', 'normalRefCount',\n",
      "       'patientId', 'proteinChange', 'proteinPosEnd', 'proteinPosStart',\n",
      "       'referenceAllele', 'refseqMrnaId', 'sampleId', 'startPosition',\n",
      "       'studyId', 'tumorAltCount', 'tumorRefCount', 'validationStatus',\n",
      "       'variantAllele', 'variantType', 'geneticEntityId', 'hugoGeneSymbol',\n",
      "       'type'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alleleSpecificCopyNumber</th>\n",
       "      <th>aminoAcidChange</th>\n",
       "      <th>center</th>\n",
       "      <th>chr</th>\n",
       "      <th>driverFilter</th>\n",
       "      <th>driverFilterAnnotation</th>\n",
       "      <th>driverTiersFilter</th>\n",
       "      <th>driverTiersFilterAnnotation</th>\n",
       "      <th>endPosition</th>\n",
       "      <th>entrezGeneId</th>\n",
       "      <th>...</th>\n",
       "      <th>startPosition</th>\n",
       "      <th>studyId</th>\n",
       "      <th>tumorAltCount</th>\n",
       "      <th>tumorRefCount</th>\n",
       "      <th>validationStatus</th>\n",
       "      <th>variantAllele</th>\n",
       "      <th>variantType</th>\n",
       "      <th>geneticEntityId</th>\n",
       "      <th>hugoGeneSymbol</th>\n",
       "      <th>type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Broad</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>98928394</td>\n",
       "      <td>200403</td>\n",
       "      <td>...</td>\n",
       "      <td>98928394</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>3</td>\n",
       "      <td>18</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>VWA3B</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Broad</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>198267491</td>\n",
       "      <td>23451</td>\n",
       "      <td>...</td>\n",
       "      <td>198267491</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>35</td>\n",
       "      <td>27</td>\n",
       "      <td>NA</td>\n",
       "      <td>G</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>SF3B1</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Broad</td>\n",
       "      <td>23</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>83128311</td>\n",
       "      <td>1538</td>\n",
       "      <td>...</td>\n",
       "      <td>83128311</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>4</td>\n",
       "      <td>42</td>\n",
       "      <td>NA</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>CYLC1</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Broad</td>\n",
       "      <td>12</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>15654580</td>\n",
       "      <td>5800</td>\n",
       "      <td>...</td>\n",
       "      <td>15654580</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>38</td>\n",
       "      <td>35</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>PTPRO</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Broad</td>\n",
       "      <td>8</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>139164553</td>\n",
       "      <td>51059</td>\n",
       "      <td>...</td>\n",
       "      <td>139164553</td>\n",
       "      <td>cll_broad_2022</td>\n",
       "      <td>3</td>\n",
       "      <td>49</td>\n",
       "      <td>NA</td>\n",
       "      <td>A</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>FAM135B</td>\n",
       "      <td>protein-coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  alleleSpecificCopyNumber aminoAcidChange center chr driverFilter  \\\n",
       "0                     None            None  Broad   2         None   \n",
       "1                     None            None  Broad   2         None   \n",
       "2                     None            None  Broad  23         None   \n",
       "3                     None            None  Broad  12         None   \n",
       "4                     None            None  Broad   8         None   \n",
       "\n",
       "  driverFilterAnnotation driverTiersFilter driverTiersFilterAnnotation  \\\n",
       "0                   None              None                        None   \n",
       "1                   None              None                        None   \n",
       "2                   None              None                        None   \n",
       "3                   None              None                        None   \n",
       "4                   None              None                        None   \n",
       "\n",
       "   endPosition  entrezGeneId  ... startPosition         studyId tumorAltCount  \\\n",
       "0     98928394        200403  ...      98928394  cll_broad_2022             3   \n",
       "1    198267491         23451  ...     198267491  cll_broad_2022            35   \n",
       "2     83128311          1538  ...      83128311  cll_broad_2022             4   \n",
       "3     15654580          5800  ...      15654580  cll_broad_2022            38   \n",
       "4    139164553         51059  ...     139164553  cll_broad_2022             3   \n",
       "\n",
       "  tumorRefCount validationStatus variantAllele  variantType  geneticEntityId  \\\n",
       "0            18               NA             A          SNP             None   \n",
       "1            27               NA             G          SNP             None   \n",
       "2            42               NA             C          SNP             None   \n",
       "3            35               NA             A          SNP             None   \n",
       "4            49               NA             A          SNP             None   \n",
       "\n",
       "  hugoGeneSymbol            type  \n",
       "0          VWA3B  protein-coding  \n",
       "1          SF3B1  protein-coding  \n",
       "2          CYLC1  protein-coding  \n",
       "3          PTPRO  protein-coding  \n",
       "4        FAM135B  protein-coding  \n",
       "\n",
       "[5 rows x 35 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data written to ../../data/cll_broad_2022_clinical_data.csv\n",
      "Data written to ../../data/cll_broad_2022_mutations.csv\n"
     ]
    }
   ],
   "source": [
    "driver()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "x = cbioportal.Clinical_Attributes.getAllClinicalAttributesInStudyUsingGET(\n",
    "    studyId=\"cll_broad_2022\",\n",
    "    projection=\"DETAILED\"\n",
    ").result()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AGE Age at diagnosis\n",
      "AGE_SAMPLING Age at sampling\n",
      "ARRAY450K Sample included in 450K methylation array \n",
      "CANCER_TYPE Cancer Type\n",
      "CANCER_TYPE_DETAILED Cancer Type Detailed\n",
      "CENTER Sequencing center\n",
      "CLL_EPITYPE CLL epitype: naive-like (n-CLL), intermediate (i-CLL) and memory-like (m-CLL)\n",
      "COHORT Cohort\n",
      "DEATH_DAYS Death (days) from the date of sampling\n",
      "DISEASE_TYPE Disease type\n",
      "EC_DISCOVERY Sample included in expression cluster discovery set (1=yes, 0=no)\n",
      "EC_EXTENSION Sample included in expression cluster extension set (1=yes, 0=no)\n",
      "EXPRESSION_CLUSTER Expression cluster\n",
      "FFS_MONTHS Failure-free survival (months)\n",
      "FFS_STATUS Failure-free survival status\n",
      "IGHV_IDENTITY_PERCENTAGE Immunoglobulin genes mutation identity percentage\n",
      "IGHV_MUTATION_STATUS Immunoglobulin genes mutation status\n",
      "IGLV3_21_R110 IGLV3-21 R110 mutation status\n",
      "M-CLL_WES Sample included in M-CLL WES genetic candidate driver discovery cohort (1=yes, 0=no)\n",
      "MUTATION_COUNT Mutation Count\n",
      "NORMAL_MEAN_COVERAGE Normal Mean Coverage (WES)\n",
      "ONCOTREE_CODE Oncotree code\n",
      "OS_MONTHS Overall survival (months) from the date of the sequenced sample\n",
      "OS_STATUS Overall survival status from the date of the sequenced sample\n",
      "PRIOR_TREATMENT_CATEGORY Prior treatment category\n",
      "RNA_SEQUENCING Sample included in RNA-Sequencing\n",
      "RRBS Sample included in reduced representation bisulfite sequencing \n",
      "SAMPLE_COUNT Number of Samples Per Patient\n",
      "SEQUENCING_PLATFORM Sequencing platform\n",
      "SEX Sex\n",
      "SOMATIC_STATUS Somatic status\n",
      "TMB_NONSYNONYMOUS TMB (nonsynonymous)\n",
      "TREATMENT_AFTER_SAMPLING First treatment after sampling\n",
      "TREATMENT_STATUS Treatment status at sampling\n",
      "TUMOR_MEAN_COVERAGE Tumor mean coverage (WES)\n",
      "TUMOR_MOLECULAR_SUBTYPE Tumor molecular subtype: mutated IGHV (M-CLLs), unmutated IGHV (U-CLLs)\n",
      "TUMOR_SAMPLE_PLOIDY Tumor sample ploidy (WES)\n",
      "TUMOR_SAMPLE_PURITY Tumor sample purity (WES)\n",
      "U-CLL_WES Sample included in U-CLL WES genetic candidate driver discovery cohort (1=yes, 0=no)\n",
      "U1_STATUS Final U1 mutational status\n"
     ]
    }
   ],
   "source": [
    "for i in x:\n",
    "    # if not i.patientAttribute:\n",
    "    print(i.clinicalAttributeId, i.description)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "studyId = \"cll_broad_2022\"\n",
    "mutations = cbioportal.Mutations.getMutationsInMolecularProfileBySampleListIdUsingGET(\n",
    "        molecularProfileId=f'{studyId}_mutations',\n",
    "        sampleListId=f'{studyId}_all',\n",
    "        projection='DETAILED'\n",
    "    ).result()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Mutation(alleleSpecificCopyNumber=None, aminoAcidChange=None, center='Broad', chr='2', driverFilter=None, driverFilterAnnotation=None, driverTiersFilter=None, driverTiersFilterAnnotation=None, endPosition=98928394, entrezGeneId=200403, gene=Gene(entrezGeneId=200403, geneticEntityId=None, hugoGeneSymbol='VWA3B', type='protein-coding'), keyword='VWA3B A1212 missense', molecularProfileId='cll_broad_2022_mutations', mutationStatus='NA', mutationType='Missense_Mutation', namespaceColumns=None, ncbiBuild='GRCh37', normalAltCount=0, normalRefCount=12, patientId='P-CRC-0001', proteinChange='A1212T', proteinPosEnd=1212, proteinPosStart=1212, referenceAllele='G', refseqMrnaId='NM_144992.4', sampleId='CRC-0001', startPosition=98928394, studyId='cll_broad_2022', tumorAltCount=3, tumorRefCount=18, uniquePatientKey='UC1DUkMtMDAwMTpjbGxfYnJvYWRfMjAyMg', uniqueSampleKey='Q1JDLTAwMDE6Y2xsX2Jyb2FkXzIwMjI', validationStatus='NA', variantAllele='A', variantType='SNP')"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutations[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "msc-thesis-BHB9aft-",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
